viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
+        if movie["title"] == title:
+            in_watchlist = True
+            break
+        movie_index += 1
+    if not in_watchlist:
+        return user_data_copy
+    movie = user_data_copy["watchlist"][movie_index]
+    del user_data_copy["watchlist"][movie_index]
+    user_data_copy = add_to_watched(user_data_copy, movie)
+    return user_data_copy
+'''
+This function takes in 1 value, a dictionary called user_data. It calculates the average rating for all the movies that are under
+the key "watched" from user_data. 
+'''
+def get_watched_avg_rating(user_data):
+    num_of_movies = len(user_data["watched"])
+    if num_of_movies == 0:
+        return 0.0
+    ratings = 0
+    for movie in user_data["watched"]:
+        ratings += movie["rating"]
+    average_rating = ratings / num_of_movies
+    return average_rating   
+'''
+This function takes in 1 value, a dictionary called user_data. It looks at all the movies that are under the key "watched"
+and figures out which genre is most watched by the user. 
+'''
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return None
+    genre_counts = {}
+    for movie in user_data["watched"]:
+        movie_genre = movie["genre"]
+        try:
+            genre_counts[movie_genre] += 1
+        except KeyError:
+            genre_counts[movie_genre] = 1
+    max_watch_count = 0
+    for genre, watch_count in genre_counts.items():
+        if watch_count > max_watch_count:
+            max_watch_count = watch_count
+            max_genre = genre
+    return max_genre
+'''
+This function takes in 1 value, a dictionary called user_data. We compare the movies under the key "watched" with the movies under
+each friend's "watched" lists. We retun a list of movies that are unique to the user. 
+'''
+def get_unique_watched(user_data):
+    unique_titles = []
+    for movie in user_data["watched"]:
+        friend_watched = False
+        for friend in user_data["friends"]:
+            for friends_movie in friend["watched"]:
+                if friends_movie["title"] == movie["title"]:
+                    friend_watched = True
+                    break
+        if not friend_watched:    
+            unique_titles.append(movie)
+    return unique_titles   
+'''
+This function takes in 1 value, a dictionary called user_data. We compare the movies under the key "watched" with the movies under
+each friend's "watched" lists. We retun a list of movies that are unique to the friends. 
+'''
+def get_friends_unique_watched(user_data):
+    unique_titles = []
+    unique_title_names = []
+    for friend in user_data["friends"]:
+        for friends_movie in friend["watched"]:
+            user_watched = False
+            for movie in user_data["watched"]:
+                if friends_movie["title"] == movie["title"]:
+                    user_watched = True
+                    break
+            if not user_watched and friends_movie["title"] not in unique_title_names:

```

Comment:
[perf] What's the big-O time complexity of *this* function? Can you think of other ways you could implement it that might reduce the complexity? Also consider what the `in` operator might contribute to complexity